src/Avalonia.Base/Platform/Storage/FileIO/BclLauncher.cs

```diff
@@ -0,0 +1,90 @@
+﻿using System;
+using System.Diagnostics;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Avalonia.Compatibility;
+using Avalonia.Metadata;
+
+namespace Avalonia.Platform.Storage.FileIO;
+
+internal class BclLauncher : ILauncher
+{
+    public virtual Task<bool> LaunchUriAsync(Uri uri)
+    {
+        _ = uri ?? throw new ArgumentNullException(nameof(uri));
+        if (uri.IsAbsoluteUri)
+        {
+            return Task.FromResult(Exec(uri.AbsoluteUri));
+        }
+
+        return Task.FromResult(false);
+    }
+
+    /// <summary>
+    /// This Process based implementation doesn't handle the case, when there is no app to handle link.
+    /// It will still return true in this case.
+    /// </summary>
+    public virtual Task<bool> LaunchFileAsync(IStorageItem storageItem)
+    {
+        _ = storageItem ?? throw new ArgumentNullException(nameof(storageItem));
+        if (storageItem.TryGetLocalPath() is { } localPath
+            && CanOpenFileOrDirectory(localPath))
+        {
+            return Task.FromResult(Exec(localPath));
+        }
+
+        return Task.FromResult(false);
+    }
+
+    protected virtual bool CanOpenFileOrDirectory(string localPath) => true;
+    
+    private static bool Exec(string urlOrFile)
+    {
+        if (OperatingSystemEx.IsLinux())
+        {
+            // If no associated application/json MimeType is found xdg-open opens return error
+            // but it tries to open it anyway using the console editor (nano, vim, other..)
+            ShellExec($"xdg-open {urlOrFile}", waitForExit: false);
+            return true;
+        }
+        else if (OperatingSystemEx.IsWindows() || OperatingSystemEx.IsMacOS())
+        {
+            using var process = Process.Start(new ProcessStartInfo
+            {
+                FileName = OperatingSystemEx.IsWindows() ? urlOrFile : "open",
+                Arguments = OperatingSystemEx.IsMacOS() ? $"{urlOrFile}" : "",
+                CreateNoWindow = true,
+                UseShellExecute = OperatingSystemEx.IsWindows()
+            });
+            return true;
+        }
+        else
+        {
+            return false;
```

Comment:
Shouldn't this throw a platform not supported exception instead of just returning false? Granted I could just be misunderstanding here what the return values mean. The XML docs on `ILauncher` don't really explain it. If it's meant to indicate if it worked or not, then I would expect the methods to be called something like `TryLaunchFile` or `AttemptToLaunchFile` to make it more explicit that it may not work.

Reply:
This API is based from UWP Laucnher which basically tells "true, if opened, false if wasn't for whatever reason" .
https://learn.microsoft.com/en-us/uwp/api/windows.system.launcher.launchuriasync?view=winrt-22621#windows-system-launcher-launchuriasync(windows-foundation-uri).

There is another LaunchUriForResultsAsync method, a newer one, which actually returns a status. But I didn't want to implement this another API model, as we would have hard time implementing all possible statuses on other platforms. Like, with `Process.Start` approach we can't even know if Uri/File was handled or not.